src/detections/rule/selectionnodes.rs

```diff
@@ -377,6 +377,12 @@ impl SelectionNode for LeafSelectionNode {
         }
 
         let event_value = self.get_event_value(event_record);
+        if self.get_key() == "EventID" && !self.select_value.is_null() {
```

Comment:
If the content of the EventID is not intended to be a String and you want to make sure it is a number, how about checking with is_number instead of is_null?

``` suggestion
        if self.get_key() == "EventID" && self.select_value.is_number() {
```